kube-client/src/client/config_ext.rs

```diff
@@ -192,7 +194,8 @@ impl ConfigExt for Config {
 
     #[cfg(feature = "openssl-tls")]
     fn openssl_ssl_connector_builder(&self) -> Result<openssl::ssl::SslConnectorBuilder> {
-        tls::openssl_tls::ssl_connector_builder(self.identity_pem().as_ref(), self.root_cert.as_ref())
+        let identity = self.exec_identity_pem().or_else(|| self.identity_pem());
```

Comment:
most in-cluster tests are currently failing, wondering if it's related to the default here being the new `exec_identity_pem` rather than the `identity_pem` fn

Reply:
The error is
```text
Error: OpensslTls(CreateSslConnector(DeserializeCertificateChain(ErrorStack([Error { code: 75497580, library: "PEM routines", reason: "no start line", file: "../crypto/pem/pem_lib.c", line: 763, data: "Expecting: CERTIFICATE" }]))))
```

I'm glad the integration tests caught this, but it's using exec auth? My local k3d user has `client-certificate-data`.

---

Maybe `clientCertificateData` in the response is an empty string, so we have `Auth::Certificate(Some(""), Some(""))`.

https://github.com/kube-rs/kube/blob/be078244da761323f8bfe93bc167db9305548207/kube-client/src/client/auth/mod.rs#L467-L470

Reply:
Hmm, does `just test` run all the same tests as GHA? I'm unable to reproduce this locally against my kind cluster which also uses `client-certificate-data`.

Reply:
`just test` only does the unit tests. this would be the `#[ignore]` marked integration tests that can be run via `just test-integration` (which will modify your current cluster). there's also a failure via `just e2e-mink8s`

Reply:
Thanks! Looks like I was running the right commands manually that I pulled from the GHA and still getting all green locally on my mac. I will see if I can get `act` working and/or test in a new environment to try to reproduce.

Reply:
I was able to get [`act`](https://github.com/nektos/act) working with the integration test but was unable to reproduce the error. Running `just test-integration` in my fresh linux environment against a `k3s` also lead to successful tests and no errors.

Reply:
hm, weirdly CI is also green now. Maybe github actions was going through a rough patch during the previous run?

Reply:
Caught some spurious k3d failures on this a few days ago (which looked either github or k3d related; failure to create the cluster). These all disappeared on later retries, and this passes consistently now.